ms2deepscore/train_new_model/data_generators.py

```diff
@@ -177,3 +181,113 @@ def _data_augmentation_spectrum(self, spectrum_tensor):
                                                 )
             spectrum_tensor[indices_noise] = self.model_settings.augment_noise_intensity * torch.rand(len(indices_noise))
         return spectrum_tensor
+
+
+### Data Generator for training an embedding evaluation model
+
+
+class DataGeneratorEmbeddingEvaluation:
+    """Generates data for training an embedding evaluation model.
```

Comment:
That's true. But somehow also a weakness of the on-for-all setting object. For us that might be handy, but I could imagine that this hides a bit of what's happening to new users. What do you think?

Reply:
Yes, good point, I see how it can be confusing. I do actually not expect many people to dive that deep into the codebase. Some will train new models, but I would expect them to use the wrapper functions that we provide. So to me, the most important people to design for is ourselves in the future. So the design that is clearest to us in 1 or 2 years. Still I do not know which of the two that is... I would use the settings in the model, or I would make a separate evaluatorSettings class. We now save them separately, but still save all the settings of the "other" models in there as well, which is a bit unexpected.